### ISSUE TITLE
Wrong cursor position

### ISSUE THREAD

**Issue by nchursin** [Wramberg/TerminalView#19]
Please provide the following information when logging a bug:

* MacOS Sierra 10.12.5
* ST3 3126
* zsh 5.2 (x86_64-apple-darwin16.0) / GNU bash, version 4.4.5(1)-release (x86_64-apple-darwin16.3.0) / oh-my-zsh is installed
* "cmd": "/bin/bash -l -c /bin/zsh",

![screen shot 2017-07-03 at 17 12 30](https://user-images.githubusercontent.com/8916229/27796522-f1aa482e-6012-11e7-88d6-b9ecb6ec82f8.png)
So basically the problem is that my cursor appears on the second line. It appears only in `zsh` and came with the latest update. 
After I hit space it moves the word I typed to the right place (first line) but keeps typing on the second line. Nothing critical but pretty annoying.

**Comment by Wramberg** [comment]
Im surprised that I have two issues now from the last update (I assume you mean from 0.3 to 0.4) since it didnt introduce any changes to terminal <-> shell interaction (been over the commit log twice now).

Did you have colors enabled before ?
Would you mind trying to checkout 0.3 manually and verifying that it works there ?
and possibly dump your env from the temrinal inside ST3 ?

**Comment by nchursin** [comment]
Yup, it was working in 0.3.0, but I'll check it anyway.
Colors were enabled before.
Here's the dump:
```
TERM=linux                                                                                                                     
SHELL=/bin/zsh                                                                                                                 
TMPDIR=/var/folders/pd/rxlrj2y5647cb4nldgq6hvtr0000gn/T/                                                                       
Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.7Vj7Q9hhgu/Render                                                    
USER=nchursin                                                                                                                  
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.a3ZH4kWFLW/Listeners                                                              
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0                                                                                          
PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/nchursin/Documents/git-tf:/opt/X11/bin                                
XPC_FLAGS=0x0                                                                                                                  
XPC_SERVICE_NAME=0                                                                                                             
SHLVL=2                                                                                                                        
HOME=/Users/nchursin                                                                                                           
LOGNAME=nchursin                                                                                                               
DISPLAY=/private/tmp/com.apple.launchd.C77wM3ftYo/org.macosforge.xquartz:0                                                     
_=/usr/bin/env                                                                                                                                          
ZSH=/Users/nchursin/.oh-my-zsh                                                                                                 
PAGER=less                                                                                                                     
LESS=-R                                                                                                                        
LSCOLORS=Gxfxcxdxbxegedabagacad                                                                                                
LC_CTYPE=   
```

**Comment by Wramberg** [comment]
Havent been able to reproduce but looking through the changes again I think it may be caused by sublime word wrap.. try turning word wrap off in the view through e.g. a keybinding

```
 { "keys": ["alt+q"], "command": "toggle_setting", "args": {"setting": "word_wrap"}},
```

I enabled wordwrap to avoid horizontal scroll bar (and because I couldnt get wrapping to happen anyway). If it happens in your setup in very curious as to what causes it but lets wait and see.

**Comment by nchursin** [comment]
Yup, that was it, thanks!
For those who have the same problem, you can use the following solution to open `TerminalView` in one command:
1. Install `Chain Of Commands` plugin
2. Use the following shortcut to open `TerminalView`:
```json
        {
		"keys": ["ctrl+alt+t"],
		"command": "chain",
		"args": {
			"commands":[
				[
					"terminal_view_open",
					{
                                                // arguments, if you use any. E.g.:
						"cmd": "/bin/bash -l -c /bin/zsh",
						"title": "Terminal (ZSH)"
					}
				],
				[
					"toggle_setting",
					{
						"setting": "word_wrap"
					}
				]
			]
		}
	}
```

**Comment by Wramberg** [comment]
Ill see if I can make time to release a hotfix later today (I specifically enable it in the plugin to avoid the horizontal scrollbar).

Guess I was wrong in assuming that wrapping would never happen if I ensure the terminal is 3 characters more narrow than the window at all times.

**Comment by rwols** [comment]
It could also be the DPI or zoom settings of the OS somehow.

**Comment by Wramberg** [comment]
I tested various DPI/resolution and zoom. Didnt seem to have any effect.

**Comment by Wramberg** [comment]
Released fix for this so wrapping is specifically turned off in the terminal: https://github.com/Wramberg/TerminalView/releases/tag/v0.4.2
